fix: render slack emoji shortcodes as unicode characters

Lomuzord opened the pull request:
This PR fixes an issue where Slack emojis are rendered as text shortcodes (e.g. :rocket:) instead of the actual emoji characters in Chatwoot messages.

It introduces a new EmojiFormatter class that uses the emoji-data mapping to convert shortcodes to unicode characters.

Lomuzord committed:
fix: render slack emoji shortcodes as unicode characters

Lomuzord committed:
Fix RuboCop ClassLength violation in EmojiFormatter

Lomuzord committed:
Refactor EmojiFormatter to load mapping from JSON file

stonecharioteer commented on lib/integrations/slack/emoji_formatter.rb:
```diff
@@ -0,0 +1,16 @@
+require 'json'
+
+class Integrations::Slack::EmojiFormatter
+  EMOJI_MAPPING = JSON.parse(File.read(File.join(__dir__, 'emojis.json'))).freeze
```
Is it possible to get the slack emojis from somewhere upstream instead?

Lomuzord committed:
refactor: use gemoji gem for slack emoji formatting

sony-mathew committed:
Merge branch 'develop' into fix/slack-emoji-rendering

sony-mathew committed:
fix: address slack emoji lint feedback

Keep the gemoji find_by_alias API while documenting the RuboCop false positive.

Revert the unrelated Bundler version churn in Gemfile.lock.

sony-mathew committed:
fix: reduce slack message creation complexity

Extract Slack message content formatting from create_message to satisfy the ABC size limit.

sony-mathew committed:
Merge branch 'develop' into fix/slack-emoji-rendering

sony-mathew reviewed (approved):
LGTM. Thank you the contribution @Lomuzord !
